src/core/reactor.ts

```diff
@@ -354,10 +354,24 @@ export class Action<T extends Present> extends ScheduledTrigger<T> implements Re
                     tag = tag.getMicroStepLater();
                 }
             }
-    
-            if (this.action.origin == Origin.logical && !(this.action instanceof Startup)) {
+
+            if (this.action.origin == Origin.logical && !(this.action instanceof Startup)
```

Comment:
I would suggest to put this in the `else` clause of the ` if (this.action instanceof FederatePortAction) {` below (and remove the last condition from the branch predicate).

Reply:
Good point. Done in https://github.com/lf-lang/reactor-ts/pull/71/commits/61498a6a4afec457cffff3a550f396856a22468a.